resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesExecutorBackend.scala

```diff
@@ -49,7 +49,7 @@ private[spark] object KubernetesExecutorBackend extends Logging {
   def main(args: Array[String]): Unit = {
     val createFn: (RpcEnv, Arguments, SparkEnv, ResourceProfile, String) =>
       CoarseGrainedExecutorBackend = { case (rpcEnv, arguments, env, resourceProfile, execId) =>
-        new CoarseGrainedExecutorBackend(rpcEnv, arguments.driverUrl, execId,
+        new KubernetesCoarseGrainedExecutorBackend(rpcEnv, arguments.driverUrl, execId,
         arguments.bindAddress, arguments.hostname, arguments.cores,
         env, arguments.resourcesFileOpt, resourceProfile)
```

Comment:
Instead of creating a new class `KubernetesCoarseGrainedExecutorBackend`, can we do the following simply?
```scala
         new CoarseGrainedExecutorBackend(rpcEnv, arguments.driverUrl, execId,
         arguments.bindAddress, arguments.hostname, arguments.cores,
-        env, arguments.resourcesFileOpt, resourceProfile)
+        env, arguments.resourcesFileOpt, resourceProfile) {
+          override def getDriverAttributes: Option[Map[String, String]] = Some(
+            super.getDriverAttributes.getOrElse(Map.empty) ++ Map(
+              "APP_ID" -> System.getenv(ENV_APPLICATION_ID),
+              "KUBERNETES_NAMESPACE" -> conf.get(KUBERNETES_NAMESPACE),
+              "KUBERNETES_POD_NAME" -> System.getenv(ENV_DRIVER_POD_NAME)))
+        }
```